crud/src/main/java/com/redhat/lightblue/crud/CRUDFindRequest.java

```diff
@@ -187,9 +187,12 @@ public void fromJson(ObjectNode node) {
                 x=node.get("maxResults");
                 if(x!=null) {
                     long l=x.asLong();
-                    if(l>=0) {
+                    if (l > 0) {
                         to=(from==null?0:from)+l-1;
```

Comment:
range is inclusive. [0,0] returns the 0'th document. [0,-1] returns nothing.

On Wed, Mar 23, 2016 at 9:07 PM, Dennis Crissman <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/crud/CRUDFindRequest.java
> <https://github.com/lightblue-platform/lightblue-core/pull/619#discussion_r57269777>
> :
>
> > @@ -187,9 +187,12 @@ public void fromJson(ObjectNode node) {
> >                  x=node.get("maxResults");
> >                  if(x!=null) {
> >                      long l=x.asLong();
> > -                    if(l>=0) {
> > +                    if (l > 0) {
> >                          to=(from==null?0:from)+l-1;
>
> @bserdar <https://github.com/bserdar> - I have done some testing and
> gotten some mixed results. Maybe it might be better to get some
> clarification.
>
> Assuming multiple documents were already inserted. A find operation with
> (from:0,to:0) should return 1 jsonDoc result or 0 results? If 1 result,
> then what is the appropriate 'to' value to set for 0 JsonDoc results
> returned? -1?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/619/files/5c1da5788283518e8e42ae36da702941ab854c41#r57269777>
>